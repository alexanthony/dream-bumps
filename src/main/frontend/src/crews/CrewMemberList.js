import React, {PropTypes} from 'react';
import { connect } from 'react-redux';
import {Card, CardTitle, CardText} from 'material-ui/Card';

import { fetchCrewMembers } from './actions';
import { getCrewMembers } from '../rootReducer';
import seatNumberToName from '../utils/seatNumberToName';


class CrewMemberList extends React.Component {
    componentWillMount() {
      if (this.props.crewId) {
        this.props.fetchCrewMembers(this.props.crewId);
      }
    }

  render() {
    if (this.props.crewMembers === undefined) {
      return null;
    }
    return (
      <Card>
        <CardTitle title="Crew Members"/>
        <CardText>
          <div>
            {
              this.props.crewMembers.map(member => (
                <dl key={member.id} className="dl-horizontal">
                  <dt>{seatNumberToName(member.seat)}</dt>
                  <dd>{member.name}</dd>
                </dl>
              ))
            }
          </div>
        </CardText>
      </Card>
    );
  }
}

CrewMemberList.propTypes = {
  crewId: PropTypes.string,
  fetchCrewMembers: PropTypes.func,
  crewMembers: PropTypes.array
};

const mapStateToProps = (state, ownProps) => ({
  crewMembers: getCrewMembers(state)(ownProps.crewId)
});

export {CrewMemberList};

export default connect(mapStateToProps, { fetchCrewMembers })(CrewMemberList);
