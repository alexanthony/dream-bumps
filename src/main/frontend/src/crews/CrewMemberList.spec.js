import React from 'react';
import { shallow } from 'enzyme';
import { CrewMemberList } from './CrewMemberList';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const fetchCrewMembers = jest.fn();
  const component = shallow(
    <CrewMemberList
      fetchCrewMembers={fetchCrewMembers}
      crewId="1"
      crewMembers={[{id: 1, seat: 1, name: 'John'}, {id: 2, seat: 2, name: 'Bob'}]}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
  expect(fetchCrewMembers).toHaveBeenCalled();
  expect(fetchCrewMembers.mock.calls.length).toBe(1);
  expect(fetchCrewMembers.mock.calls[0][0]).toBe('1');
});
