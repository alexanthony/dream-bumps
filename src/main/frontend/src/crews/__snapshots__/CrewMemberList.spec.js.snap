exports[`test Renders properly 1`] = `
<Card
  expandable={false}
  expanded={null}
  initiallyExpanded={false}>
  <CardTitle
    title="Crew Members" />
  <CardText>
    <div>
      <dl
        className="dl-horizontal">
        <dt>
          Bow
        </dt>
        <dd>
          John
        </dd>
      </dl>
      <dl
        className="dl-horizontal">
        <dt>
          2
        </dt>
        <dd>
          Bob
        </dd>
      </dl>
    </div>
  </CardText>
</Card>
`;
